lua/class_definitions.lua

```diff
@@ -75,6 +75,24 @@ classes = {
         functions = {
         }
     },
+    calendar = {
+        attributes = {
+        },
+        functions = {
+            { name = "get_turn", rval = "int", args = { } },
+            { name = "minutes_past_midnight", rval = "int", args = { } },
+            { name = "seconds_past_midnight", rval = "int", args = { } },
+            { name = "sunrise", rval = "calendar", args = { } },
+            { name = "sunset", rval = "calendar", args = { } },
+            { name = "is_night", rval = "bool", args = { } },
+            { name = "sunlight", rval = "calendar", args = { } },
```

Comment:
The by-ref vs by-value thing is (currently) part of the type. If type X is defined as by-value, all usage of that type will be by-value. A per-function setting is not supported (feel free to change this).